dotnet/src/Functions/Functions.OpenApi.Extensions/Extensions/OpenApiKernelExtensions.cs

```diff
@@ -0,0 +1,160 @@
+            var apiDependencyDetails = apiDependency.Value;
+
+            var apiDescriptionUrl = apiDependencyDetails.ApiDescriptionUrl;
+
+            var openApiDocumentString = await DocumentLoader.LoadDocumentFromUriAsync(new Uri(apiDescriptionUrl),
+                kernel.LoggerFactory.CreateLogger(typeof(OpenApiKernelExtensions)) ?? NullLogger.Instance,
+                httpClient,
+                authCallback: null,
+                executionParameters?.UserAgent,
+                cancellationToken).ConfigureAwait(false);
+
+            OpenApiDiagnostic diagnostic = new();
+            var openApiDocument = new OpenApiStringReader(new()
+            {
+                BaseUrl = new(apiDescriptionUrl)
+            }
+            ).Read(openApiDocumentString, out diagnostic);
+
+            var requestUrls = new Dictionary<string, List<string>>(StringComparer.OrdinalIgnoreCase);
+            var pathMethodPairs = apiDependencyDetails.Requests.Select(request => (request.UriTemplate, request.Method?.ToUpperInvariant()));
+            foreach (var (UriTemplate, Method) in pathMethodPairs)
+            {
+                if (UriTemplate is null || Method is null)
+                {
+                    continue;
+                }
+
+                if (requestUrls.TryGetValue(UriTemplate, out List<string>? value))
+                {
+                    value.Add(Method);
+                    continue;
+                }
+
+                requestUrls.Add(UriTemplate, new List<string>() { Method });
+            }
+
+            var predicate = OpenApiFilterService.CreatePredicate(null, null, requestUrls, openApiDocument);
+            var filteredOpenApiDocument = OpenApiFilterService.CreateFilteredDocument(openApiDocument, predicate);
+            ILoggerFactory loggerFactory = kernel.LoggerFactory;
+            var logger = loggerFactory.CreateLogger(typeof(OpenApiKernelExtensions));
+
+            var serverUrl = filteredOpenApiDocument.Servers.FirstOrDefault()?.Url;
+
+            var runner = new RestApiOperationRunner(
+                httpClient,
+                executionParameters?.AuthCallback,
+                executionParameters?.UserAgent,
+                executionParameters?.EnableDynamicPayload ?? true,
+                executionParameters?.EnablePayloadNamespacing ?? false);
+
+            foreach (var path in filteredOpenApiDocument.Paths)
+            {
+                var operations = OpenApiDocumentParser.CreateRestApiOperations(serverUrl, path.Key, path.Value);
+                foreach (RestApiOperation operation in operations)
+                {
+                    try
+                    {
+                        logger.LogTrace("Registering Rest function {0}.{1}", pluginName, operation.Id);
+                        functions.Add(Microsoft.SemanticKernel.Plugins.OpenApi.OpenApiKernelExtensions.CreateRestApiFunction(pluginName, runner, operation, executionParameters, new Uri(serverUrl), loggerFactory));
```

Comment:
Consider moving the 'Microsoft.SemanticKernel.Plugins.OpenApi.OpenApiKernelExtensions' namespace to the 'using' directive.

Reply:
I had to use fully qualified name due to name collision of `OpenApiKernelExtensions` in both namespaces. I renamed this file to `ApiManifestKernelExtensions`.